eppo_core/src/ufc/eval_details.rs

```diff
@@ -0,0 +1,102 @@
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
+use crate::{AttributeValue, Attributes};
+
+use super::{AssignmentValue, Condition, FlagEvaluationError, Shard, Value};
+
+/// Details about feature flag evaluation.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EvalFlagDetails {
+    pub flag_key: String,
+    pub subject_key: String,
+    pub subject_attributes: Attributes,
+    /// Timestamp when the flag was evaluated.
+    pub timestamp: DateTime<Utc>,
+    /// Details of configuration used for evaluation. None if configuration hasn't been fetched yet.
+    pub configuration_details: Option<ConfigurationDetails>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub result: Option<AssignmentValue>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub error: Option<FlagEvaluationError>,
+    /// Key of the selected variation.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub variation_key: Option<String>,
+    /// Value of the selected variation. Could be `None` if no variation is selected, or selected
+    /// value is absent in configuration (configuration error).
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub variation_value: Option<Value>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub allocations: Vec<EvalAllocationDetails>,
```

Comment:
This field contains all the information from the next fields in one array:
```ts
matchedAllocation: AllocationEvaluation | null;
unmatchedAllocations: Array<AllocationEvaluation>;
unevaluatedAllocations: Array<AllocationEvaluation>;
```
I'm personally not convinced that splitting it is more convenient. One reason is that it somewhat destroys the (perception of) evaluation order. The other one is that I assume that users more often debug why their rule doesn't match when it should, and this may happen because (1) another rule matched before it (order or specific condition is important), or (2) the condition didn't work in the way user expected. (Not saying that these are hard to debug with JS structure but IMO it doesn't help in a significant way.)

Splitting allocations here also sets a precedent and we would probably need to split `matchedRule`/`unmatchedRules`/`unevaluatedRules`, `matchedSplit`/`unmatchedSplits`/`unevaluatedSplits`, and `matchedConditions`/`unmatchedCondition`.

Reply:
Well you're conflating two things -- one is matchedAllocation, which is what I think we should add for consistency, and the other is matchedRule, which I'm okay with leaving out for these reasons. That said, I'm okay with leaving this as is if consistency between SDKs isn't important. I'd like to better understand the customer impact of diverging from the docs and if this causes support volume, so I sent this to the project channel to get a better pulse on what the team thinks.

Reply:
Sounds good.

I understand that you're not asking to split rules etc. What I'm saying is that from the API design perspective, it's weird to split in one place but not others.

Reply:
Sounds like we can move forward with this so long as SDK-level documentation is added for this feature.